src/components/leaderboard/LeaderboardContext.tsx

```diff
@@ -0,0 +1,33 @@
+import { createContext, useContext, useState } from 'react'
+import { LeaderboardRole } from '../../services/datahub/leaderboard/index'
+
+export type LeaderboardContextState = {
+  leaderboardRole: LeaderboardRole
+  setLeaderboardRole: (role: LeaderboardRole) => void
+}
+
+const LeaderboardContext = createContext<LeaderboardContextState>({} as any)
+
+type ContextWrapperProps = {
+  children: React.ReactNode
+}
+
+export const LeaderboardContextWrapper: React.FC<ContextWrapperProps> = ({
```

Comment:
Maybe better just use URL for changing it?

Context only for one string, idk

Like "role=staker | creator"